javav2/usecases/topics_and_queues/src/main/java/com/example/sns/SNSWorkflow.java

```diff
@@ -0,0 +1,669 @@
+        try {
+            List<DeleteMessageBatchRequestEntry> entries = new ArrayList<>();
+            for (Message msg: messages) {
+                DeleteMessageBatchRequestEntry entry = DeleteMessageBatchRequestEntry.builder()
+                    .id(msg.messageId())
+                    .build();
+
+                entries.add(entry);
+            }
+
+            DeleteMessageBatchRequest deleteMessageBatchRequest = DeleteMessageBatchRequest.builder()
+                .queueUrl(queueUrl)
+                .entries(entries)
+                .build();
+
+            sqsClient.deleteMessageBatch(deleteMessageBatchRequest);
+            System.out.println("The batch delete of messages was successful");
+
+        } catch (SqsException e) {
+            System.err.println(e.awsErrorDetails().errorMessage());
+            System.exit(1);
+        }
+    }
+
+    public static List<Message> receiveMessages(SqsClient sqsClient, String queueUrl, String msgAttValue) {
+        try {
+            if (msgAttValue.isEmpty()) {
+                ReceiveMessageRequest receiveMessageRequest = ReceiveMessageRequest.builder()
+                    .queueUrl(queueUrl)
+                    .maxNumberOfMessages(5)
+                    .build();
+                return sqsClient.receiveMessage(receiveMessageRequest).messages();
+            } else {
+                // We know there are filters on the message.
+                ReceiveMessageRequest receiveRequest = ReceiveMessageRequest.builder()
+                    .queueUrl(queueUrl)
+                    .messageAttributeNames(msgAttValue) // Include other message attributes if needed.
+                    .maxNumberOfMessages(5)
+                    .build();
+
+                return sqsClient.receiveMessage(receiveRequest).messages();
+            }
+
+        } catch (SqsException e) {
+            System.err.println(e.awsErrorDetails().errorMessage());
+            System.exit(1);
+        }
+        return null;
+    }
+
+    public static void pubMessage(SnsClient snsClient, String message, String topicArn) {
+        try {
+            PublishRequest request = PublishRequest.builder()
+                .message(message)
+                .topicArn(topicArn)
+                .build();
+
+            PublishResponse result = snsClient.publish(request);
+            System.out.println(result.messageId() + " Message sent. Status is " + result.sdkHttpResponse().statusCode());
+
+        } catch (SnsException e) {
+            System.err.println(e.awsErrorDetails().errorMessage());
+            System.exit(1);
+        }
+    }
+
+    public static void pubMessageFIFO(SnsClient snsClient, String message, String topicArn, String msgAttValue, String duplication) {
+        try {
+            PublishRequest request;
+            // Means the user did not choose to use a message attribute.
+            UUID uuid = UUID.randomUUID();
```

Comment:
FIFO only works with messages that have the same group ID. This would be better to just use the same group ID. You can use "1". I allowed the user to enter a group ID and a deduplication ID.
https://docs.aws.amazon.com/sns/latest/dg/fifo-message-grouping.html
 Also, if they selected content-based deduplication, setting the messageDeduplicationId overrides content-based deduplication.

Reply:
I will update the code to let user enter a value for both.